x/auth/client/cli/tx_sign.go

```diff
@@ -119,7 +119,10 @@ func makeSignBatchCmd() func(cmd *cobra.Command, args []string) error {
 					return err
 				}
 			} else {
-				err = authclient.SignTxWithSignerAddress(txFactory, clientCtx, multisigAddr, clientCtx.GetFromName(), txBuilder, true)
+				if txFactory.SignMode() == signing.SignMode_SIGN_MODE_UNSPECIFIED {
+					txFactory = txFactory.WithSignMode(signing.SignMode_SIGN_MODE_LEGACY_AMINO_JSON)
```

Comment:
This is a bit confusing for me. Why don't we use DIRECT as default sign mode?

Reply:
Don't we need multisig to use legacy mode for signing?

Reply:
I believe that's incorrect, the `multisig.LegacyAminoPubKey` can generate nested signatures that are SIGN_MODE_DIRECT.

Here's how I tested it: in the following test file
- https://github.com/cosmos/cosmos-sdk/blob/90e9370bd80d9a3d41f7203ddb71166865561569/crypto/keys/multisig/multisig_test.go#L273
- this creates sigs that are SIGN_MODE_UNSPECIFIED, which I believe defaults to SIGN_MODE_DIRECT
- however, in any case, if I explicitly put `SignMode: signing.SignMode_SIGN_MODE_DIRECT`, all tests still pass.

Reply:
We cannot use sign mode direct with multisig as we'll need to know all the signers info before, this may not be a problem with tests though

Reply:
We should revisit the sign modes discussion from #6078. I'm partly responsible for this, but I did also try to present alternatives, none of them got enough buy-in so we are where we are with `SIGN_MODE_DIRECT` and it's less than ideal for multisigs

Reply:
See specifically the discussion starting from here: https://github.com/cosmos/cosmos-sdk/issues/6078#issuecomment-628026933

Reply:
@zmanian could you advise on this please?

Reply:
Maybe we need a new `SIGN_MODE`?

Reply:
IMO yeah SIGN_MODE_DIRECT isn't great for multisig users but I don't think it is a show stopped.

Reply:
I suggest we merge this PR as is, to fix the existing multisig implementation of signbatch.

I'll create a separate issue to continue this discussion on a new SIGN_MODE for multisigs , which will be a breaking change and can't be incorporated into stargate at this point.